src/simulation.jl

```diff
@@ -10,11 +10,11 @@ function simulate_ogata(
     t = tmin
     while t < tmax
         B, L = ground_intensity_bound(pp, t + eps(t), h)
-        τ = B > 0 ? rand(rng, Exponential(inv(B))) : Inf
+        τ = B > 0 ? rand(rng, Exponential(inv(B))) : typemax(inv(B))
```

Comment:
```suggestion
        τ = B > 0 ? rand(rng, Exponential(T(inv(B)))) : typemax(T)
```
We already have the right type at our disposal: it is `T`

Reply:
Aren't any chance that `T(inv(B))` throws an error ?
For instance, if `T == Int`... I agree that would not be the case in practice but it is at least possible.

Reply:
No because inv(2) returns a floating point number, it doesn't throw an error

Reply:
My example is:
- assume `T=Int` and `B=2`.
- Then, `inv(2) = .5 ::Float64` and `T(inv(B))`, i.e. `Int(.5)` throws an error.

Reply:
Oh right. But then the code would error anyway because we would try to create an array of times with element type `Int` and then push real numbers into it. So I would say this doesn't change much

Reply:
Actually you're right, we can keep `inv(B)` here. The conversion will happen anyway if necessary and it's more elegant